source/common/upstream/outlier_detection_impl.cc

```diff
@@ -152,5 +164,49 @@ void DetectorImpl::runCallbacks(HostPtr host) {
   }
 }
 
+void EventLoggerImpl::logEject(HostDescriptionPtr host, EjectionType type) {
+  // TODO: Log friendly host name (e.g., instance ID or DNS name).
+  // clang-format off
+  static const std::string json =
```

Comment:
That includes newlines which need to be trimmed. I didn't want to bother with that.